core/src/lib/auth/internal/CoreStitchAuth.ts

```diff
@@ -136,8 +139,15 @@ export default abstract class CoreStitchAuth<TStitchUser extends CoreStitchUser>
     stitchReq: StitchAuthDocRequest
   ): Promise<any> {
     return this.doAuthenticatedJSONRequestRaw(stitchReq)
-      .then(response => response.body)
-      .catch(err => new StitchClientException(err));
+      .then(response => response.body) 
+      .catch(err => {
+        if (err instanceof StitchException) {
+          throw err;
+        }
+        // TODO: this may merge weirdly with Jason's PR, this is a note to self to check to make sure 
+        // it still looks correct
+        throw new StitchRequestException(err, StitchRequestErrorCode.DECODING_ERROR)
```

Comment:
We need the encoding error too as java has it but they may need to wait on Jason